spec/controllers/v0/sessions_controller_spec.rb

```diff
@@ -43,15 +43,39 @@
                             decrypted_document: response_xml_stub)
   end
   let(:saml_response_too_late) do
-    double('saml_response', is_valid?: false, status_message: '', in_response_to: uuid,
+    double('saml_response', is_valid?: false,
+                            status_message: 'Current time is on or after NotOnOrAfter condition',
+                            in_response_to: uuid,
                             is_a?: true,
                             errors: ['Current time is on or after NotOnOrAfter ' \
                               'condition (2017-02-10 17:03:40 UTC >= 2017-02-10 17:03:30 UTC)'],
                             decrypted_document: response_xml_stub)
   end
   # "Current time is earlier than NotBefore condition #{(now + allowed_clock_drift)} < #{not_before})"
   let(:saml_response_too_early) do
-    double('saml_response', is_valid?: false, status_message: '', in_response_to: uuid,
+    double('saml_response', is_valid?: false,
+                            status_message: 'Current time is earlier than NotBefore condition',
+                            in_response_to: uuid,
+                            is_a?: true,
+                            errors: ['Current time is earlier than NotBefore ' \
+                              'condition (2017-02-10 17:03:30 UTC) < 2017-02-10 17:03:40 UTC)'],
+                            decrypted_document: response_xml_stub)
+  end
+
+  let(:saml_response_data_inconsistent) do
```

Comment:
for codes 004 and 005 (MVI related) I'm making an assumption that the status message will be in the saml_response but I really don't have a solid understanding of this failure scenario to know if that's true or not @saneshark can you take a look and see if this seems ok to you?

Reply:
No, the status message will not be in the saml response in these scenarios. Instead you will want to look at the logic around how Session, User, and UserIdentity get persisted. In the case of User there are certain validations that are only run when the user is LOA3. I would characterize these as RedisStore persistence errors. They could be attribute to any of the three models failing to save (but usually attributed to MVI being down.)